src/providers/baremetal.rs

```diff
@@ -0,0 +1,221 @@
+use crate::ssh;
+use failure::Error;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+/// Descriptor for a single, existing machine to connect to.
+/// Therefore, the `impl MachineSetup` includes the address of the machine in `region`; i.e.,
+/// each instance of Setup corresponds to a single machine.
+#[derive(Clone)]
+pub struct Setup {
+    addr: std::net::SocketAddr,
+    username: String,
+    key_path: Option<std::path::PathBuf>,
+    setup_fn:
+        Option<Arc<dyn Fn(&mut ssh::Session, &slog::Logger) -> Result<(), Error> + Send + Sync>>,
+}
+
+impl super::MachineSetup for Setup {
+    type Region = String;
+    fn region(&self) -> Self::Region {
+        format!("bare:{}", self.addr)
+    }
+}
+
+impl PartialEq for Setup {
```

Comment:
Why are these impls needed?

Reply:
yep, forgot to remove, they were needed in an old version